src/blr/cli/buildmolecules.py

```diff
@@ -63,12 +57,35 @@ def main(args):
         write_molecule_stats(bc_to_mol_dict)
 
 
+def parse_reads(pysam_openfile, barcode_tag, summary):
+    for read in tqdm(pysam_openfile.fetch(until_eof=True)):
+        is_good = True
+        summary["Total reads"] += 1
+        if read.is_duplicate:
+            summary["Duplicates"] += 1
+            is_good = False
+
+        barcode = utils.get_bamtag(pysam_read=read, tag=barcode_tag)
+        if not barcode:
+            summary["Reads without barcode"] += 1
+            is_good = False
+
+        if read.is_unmapped and barcode:
+            summary[f"Unmapped {barcode_tag} tagged read"] += 1
+            is_good = False
```

Comment:
A read which isn't mapped but has a barcode will not be caught here, is there a point to that?

Reply:
This is just a different way of writing the [same code that was before](https://github.com/FrickTobias/BLR/blob/master/src/blr/cli/buildmolecules.py#L90), see below:

```
if not barcode:
            summary["Reads without barcode"] += 1
        elif read.is_unmapped:
            summary[f"Unmapped {barcode_tag} tagged read"] += 1
```

As for is this statistic is any useful I have not thought much about. Probably it is sufficient to keep track of the unmapped reads.

Reply:
Ah right. But then I'd change it since we can't see the point of keeping it that way anymore.